### PR TITLE
#160617676 liking and disliking articles

### PR DESCRIPTION
**Pull request by nakiwuge**
#### What does this PR do?
- Ensures that  authenticated  users can like and dislike articles
#### Description of Task to be completed?
- Ensure users can like and dislike articles both authored by them and an authored by them.
- Ensure users can update their likes and dislikes on a certain article
#### How should this be manually tested?
- After creating an article go to api/articles/slug/liking as shown in the screenshots below
#### What are the relevant pivotal tracker stories?
#160617676
#### Screenshots
screen shot 2018-10-12 at 12 04 43

screen shot 2018-10-12 at 12 52 02
screen shot 2018-10-12 at 12 52 46
screen shot 2018-10-12 at 12 53 29

### CODE REVIEW

**Review by PatrickCmd — COMMENTED** [review]
@zellymiriam You do have typo **disking** in your PR name, please can you correct that.

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -62,6 +62,71 @@ def update(self, request, *args, **kwargs):
...
+            return Response({"msg": "you have already rated this article"})
+        rating = request.data.get('rate', {})
+        serializer = self.serializer_class(data=rating)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer, article_slug)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def perform_create(self, serializer, article_slug):
+        serializer.save(rated_by=self.request.user, article=article_slug)
+class LikeArticleView(ListCreateAPIView):
```
Comment (PatrickCmd) [inline authors/apps/articles/views.py:89]: @zellymiriam Best practice to have two blank lines before class definition. See PEP8 style guidelines.

File: authors/apps/articles/urls.py [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
-    path('articles/<str:slug>/', ArticleAPIDetailsView.as_view(),name='retrieveUpdateDelete'),
-    path('articles/<str:slug>/rating/', RateArticleView.as_view())
-]
\ No newline at end of file
+    path('articles/<str:slug>/', ArticleAPIDetailsView.as_view(),
+         name='retrieveUpdateDelete'),
+    path('articles/<str:slug>/rating/', RateArticleView.as_view()),
+    path('articles/<str:slug>/liking/',
+         LikeArticleView.as_view(), name='like_article'),
+    path('articles/<str:slug>/liking/modify/',
```
Comment (PatrickCmd) [inline authors/apps/articles/urls.py]: @zellymiriam I think this url path would make more sense when we have it like this **path('articles/<str:slug>/dislike')**,

File: authors/apps/articles/urls.py [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 urlpatterns = [
     path("articles/", ArticleAPIView.as_view()),
-    path('articles/<str:slug>/', ArticleAPIDetailsView.as_view(),name='retrieveUpdateDelete'),
-    path('articles/<str:slug>/rating/', RateArticleView.as_view())
-]
\ No newline at end of file
+    path('articles/<str:slug>/', ArticleAPIDetailsView.as_view(),
+         name='retrieveUpdateDelete'),
+    path('articles/<str:slug>/rating/', RateArticleView.as_view()),
+    path('articles/<str:slug>/liking/',
```
Comment (PatrickCmd) [inline authors/apps/articles/urls.py]: @zellymiriam I think **like** is a better tense than **liking**

Comment (nakiwuge) [reply authors/apps/articles/urls.py]: Thanks, @PatrickCmd let me modify this.

File: authors/apps/articles/urls.py (hunk 2 above)
Comment (nakiwuge) [reply authors/apps/articles/urls.py]: Thanks, @PatrickCmd let me modify this.

File: authors/apps/articles/views.py (hunk 1 above)
Comment (nakiwuge) [reply authors/apps/articles/views.py:89]: Thanks, @PatrickCmd let me modify this.

**Review by PatrickCmd — APPROVED** [review]

**Pull request merged**